tests/beaker_tests/ntp_server/test_ntp_server.rb

```diff
@@ -0,0 +1,98 @@
+###############################################################################
+# Copyright (c) 2017-2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+# Test hash top-level keys
+tests = {
+  agent:         agent,
+  master:        master,
+  intf_type:     'ethernet',
+  resource_name: 'ntp_server',
+}
+
+intf = find_interface(tests)
+
+# Skip -ALL- tests if a top-level platform/os key exludes this platform
+skip_unless_supported(tests)
+
+tests[:default] = {
+  desc:           '1.1 Default Properties',
+  title_pattern:  '5.5.5.5',
+  manifest_props: {
+  },
+  code:           [0, 2],
+}
+
+tests[:default_ipv6] = {
+  desc:           '1.2 IPV6 Default Properties',
+  title_pattern:  '2002::5',
+  manifest_props: {
+  },
+  code:           [0, 2],
+}
+
+tests[:non_default] = {
```

Comment:
Running the tests locally I'm seeing:

```
* TestStep :: 2.1 Non-Default Properties :: RESOURCE
    test_resource :: cmd:
    bundle exec puppet device --verbose --trace --strict=error --modulepath spec/fixtures/modules --deviceconfig spec/fixtures/acceptance-device.conf --target sut --libdir lib/ --resource ntp_server'5.5.5.5'
    TestStep :: Match (?-mix:key +=> +'?1'?) :: PASS
    TestStep :: Match (?-mix:maxpoll +=> +'?7'?) :: PASS
    TestStep :: Match (?-mix:minpoll +=> +'?5'?) :: PASS
    TestStep :: Match (?-mix:prefer +=> +'?true'?) :: PASS
    stdout:
    --

    --
    Warning:
    TestStep :: Match (?-mix:source_interface +=> +'?ethernet1\/10'?) :: FAIL
```

Reply:
AH yes, my bad, it is dependent on #31 being merged first and this being rebased off it